acceptance_tests/features/steps/create_message_internal.py

```diff
@@ -2,25 +2,28 @@
 
 from acceptance_tests import browser
 from acceptance_tests.features.pages import create_message_internal
+from common.collection_exercise_utilities import find_case_by_enrolment_code
 
 
 @given("The internal user has found the associated respondent")
-def user_has_found_respondent(_):
-    assert create_message_internal.RESPONDENT_DETAILS.get_respondent_id() is not None
-    assert create_message_internal.RESPONDENT_DETAILS.get_ru_id() is not None
-    assert create_message_internal.RESPONDENT_DETAILS.get_ru_ref() is not None
+def user_has_found_respondent(context):
+    context.party_id = find_case_by_enrolment_code(context.iac)['partyId']
+
+    assert context.respondent_user_name is not None
+    assert context.party_id is not None
+    assert context.short_name is not None
 
 
 @when("they choose to send them a secure message and navigated to the 'send message' page")
```

Comment:
use of quotation marks inside given, then and when clauses should be restricted to passing variables otherr wise it is confusing e.g should be 
either 
@when("they choose to send them a secure message and navigated to the send message page")
when no params passed

or 
When("they choose to send them a secure message and navigated to the 'send message' page")
@when(@when("they choose to send them a secure message and navigated to the {page_name} page")
def navigate_to_send_message(_, page_name):     # Passes value "send message" in the page_name variable
.....

When using the quotations to send a variable into the function/clause

Reply:
fixed